pimd/pim6_cmd.c

```diff
@@ -0,0 +1,429 @@
+      IPV6_STR
+      "Generic multicast configuration options\n"
+      "Define multicast boundary\n"
+      "Filter OIL by group using prefix list\n"
+      "Prefix list to filter OIL with\n")
+{
+	nb_cli_enqueue_change(vty, "./multicast-boundary-oil", NB_OP_MODIFY,
+			argv[4]->arg);
+
+	return nb_cli_apply_changes(vty,
+			FRR_PIM_INTERFACE_XPATH,
+			"frr-routing:ipv6");
+
+}
+
+DEFUN(interface_no_ipv6_pim_boundary_oil,
+      interface_no_ipv6_pim_boundary_oil_cmd,
+      "no ipv6 multicast boundary oil [WORD]",
+      NO_STR
+      IPV6_STR
+      "Generic multicast configuration options\n"
+      "Define multicast boundary\n"
+      "Filter OIL by group using prefix list\n"
+      "Prefix list to filter OIL with\n")
+{
+	nb_cli_enqueue_change(vty, "./multicast-boundary-oil", NB_OP_DESTROY,
+			NULL);
+
+	return nb_cli_apply_changes(vty,
+			FRR_PIM_INTERFACE_XPATH,
+			"frr-routing:ipv6");
+}
+
+DEFUN (interface_ipv6_mroute,
+       interface_ipv6_mroute_cmd,
+       "ipv6 mroute INTERFACE X:X::X:X [X:X::X:X]",
+       IPV6_STR
+       "Add multicast route\n"
+       "Outgoing interface name\n"
+       "Group address\n"
+       "Source address\n")
+{
+	int idx_interface = 2;
+	int idx_ipv6 = 3;
+	const char *source_str;
+
+	if (argc == (idx_ipv6 + 1))
+		source_str = "::";
+	else
+		source_str = argv[idx_ipv6 + 1]->arg;
+
+	nb_cli_enqueue_change(vty, "./oif", NB_OP_MODIFY,
+			argv[idx_interface]->arg);
+
+	return nb_cli_apply_changes(vty,
+			FRR_PIM_MROUTE_XPATH,
+			"frr-routing:ipv6", source_str,
+			argv[idx_ipv4]->arg);
+}
+
+DEFUN (interface_no_ipv6_mroute,
+       interface_no_ipv6_mroute_cmd,
+       "no ipv6 mroute INTERFACE X:X::X:X [X:X::X:X]",
+       NO_STR
+       IPV6_STR
+       "Add multicast route\n"
+       "Outgoing interface name\n"
+       "Group Address\n"
+       "Source Address\n")
+{
+	int idx_ipv6 = 4;
+	const char *source_str;
+
+	if (argc == (idx_ipv6 + 1))
+		source_str = "::";
+	else
+		source_str = argv[idx_ipv6 + 1]->arg;
+
+	nb_cli_enqueue_change(vty, ".", NB_OP_DESTROY, NULL);
+
+	return nb_cli_apply_changes(vty,
+			FRR_PIM_MROUTE_XPATH,
+			"frr-routing:ipv6", source_str,
+			argv[idx_ipv4]->arg);
+}
+
+void pim_cmd_init(void)
+{
+	if_cmd_init(pim6_interface_config_write);
+
+	install_element(INTERFACE_NODE, &interface_ipv6_pim_cmd);
+	install_element(INTERFACE_NODE, &interface_no_ipv6_pim_cmd);
+	install_element(INTERFACE_NODE, &interface_ipv6_pim_drprio_cmd);
+	install_element(INTERFACE_NODE, &interface_no_ipv6_pim_drprio_cmd);
+	install_element(INTERFACE_NODE, &interface_ipv6_pim_hello_cmd);
+	install_element(INTERFACE_NODE, &interface_no_ipv6_pim_hello_cmd);
+	install_element(INTERFACE_NODE, &interface_ipv6_pim_activeactive_cmd);
+	install_element(INTERFACE_NODE, &interface_ipv6_pim_ssm_cmd);
+	install_element(INTERFACE_NODE, &interface_no_ipv6_pim_ssm_cmd)
+        install_element(INTERFACE_NODE, &interface_ip_pim_sm_cmd);
+        install_element(INTERFACE_NODE, &interface_no_ip_pim_sm_cmd);
```

Comment:
We use tabs and not spaces. This clearly demonstrates that the style tool is not being used. Just use clang-format with the instructions in the developer's documentation. Pointing out basic style errors like this should be unnecessary by this point.

Reply:
resolved